### PR TITLE
feat(Modal): create the modal component

### PR DESCRIPTION
**Pull request by ArthusLiang**
also include modal-header modal-footer and modal-body

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/collab-ui/collab-ui-angular/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [ ] Feature
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/lib/modal-body/modal-body.component.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    <ng-content></ng-content>
+  `,
+  styles: []
+})
+export class ModalBodyComponent implements OnInit {
+
+  /** @option css class names | '' | '' */
+  @Input() public class: string = '';
+
+  @HostBinding('class') get className(): string {
```
Comment (mohammedzamakhan) [inline src/lib/modal-body/modal-body.component.ts]: Wondering if this be much cleaner?
```ts
@HostBinding('class') className = (this.class && ` ${this.class}`) || '';
@HostBinding('class.cui-modal__body') defaultClass = true;
```

File: src/lib/modal-footer/modal-footer.component.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      <ng-content></ng-content>
+  `,
+  styles: []
+})
+export class ModalFooterComponent implements OnInit {
+
+  /** @option css class names | '' | '' */
+  @Input() public class: string = '';
+
+  @HostBinding('class') get className(): string {
```
Comment (mohammedzamakhan) [inline src/lib/modal-footer/modal-footer.component.ts]: Same here?

File: src/lib/modal-header/modal-header.component.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  @Input() public class: string = '';
+  /** ModalHeader label text | '' */
+  @Input() public headerLabel: string = '';
+  /** Modal message | '' */
+  @Input() public message: string = '';
+  /** show/hide close button | true */
+  @Input() public showCloseButton: boolean = true;
+
+  @HostBinding('class') get className(): string {
+    return 'cui-modal__header' +
```
Comment (mohammedzamakhan) [inline src/lib/modal-header/modal-header.component.ts]: Same here

File: src/lib/modal/modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import { TemplatePortal } from '@angular/cdk/portal';
+import { ModalServiceService } from './modal-service.service';
+
+export type SizeType = 'large' | 'medium' | 'default' | 'small' | 'full' | 'dialog';
+
+@Component({
+  selector: 'cui-modal',
+  template: `
+    <ng-template #overlayGlobalTemplate>
+      <div [class]="classes">
```
Comment (mohammedzamakhan) [inline src/lib/modal/modal.component.ts]: Use ngClass, similar to what @daanishnasir used here https://github.com/collab-ui/collab-ui-angular/pull/11/files#diff-374f90f152f2a50c1ad12540b901ccbfR182

File: src/lib/modal-body/modal-body.component.ts (hunk 1 above)
Comment (ArthusLiang) [reply src/lib/modal-body/modal-body.component.ts]: Hi, thanks for the suggestion. I tested this code, but it does not work currently.

Comment (mohammedzamakhan) [reply src/lib/modal-body/modal-body.component.ts]: I think this is much better pattern https://stackblitz.com/edit/angular-rrziff?file=src%2Fapp%2Fhello.component.ts

I don't know why `HostBinding` is so different that `host`

**Comment by ArthusLiang** [comment]
Create a new brunch to do this

**Pull request closed without merging**